src/hyperframe/exceptions.py

```diff
@@ -7,9 +7,20 @@
 """
 
 
-class UnknownFrameError(ValueError):
+class HyperframeError(Exception):
```

Comment:
@Kriechi would it make sense for `HyperframeError` to extend `ValueError`? I guess it comes down to whether `InvalidDataError` is a type of `ValueError`?

Reply:
I was going back on forth on this myself... in the end my idea was: let's do a major release - so let's break everything and clean this up. 

Having ValueError is really a historical burden - so I rather introduce a new parent class (same concept is already implemented in hyperframe and wsproto). A clean library project should only ever raise it's own exception type (and sub-classes of it). ValueError can mean anything and nothing.

Thoughts?

Reply:
I don't think we should break compatibility, and instead extend `ValueError`. I think this as I don't think there is much effort to follow h2 at the moment. For example, I tried to fix an [issue in Twisted](https://github.com/twisted/twisted/pull/1179) some time ago and it is still not merged (I need to figure out the Twisted testing setup). Maybe note it as deprecated in the docs and try and fix the major dependents first?

Reply:
I fully understand your concerns here - if this would be a similar in h2 itself I would be more cautious for sure.
In this case however, I believe this is a no-issue. hyperframe downloads correlate with h2 downloads pretty much 1:1 - which leads me to the conclusion there are no major dependents who directly interface with hyperframe alone.
https://pypistats.org/packages/h2 https://pypistats.org/packages/hyperframe

For hyper-h2, there is already a PR addressing the breaking change, and I would be merging everything and cut the major releases as indicated. mitmproxy, the only other dependant I know if, does not rely on such deep implementation detail.

@pgjones are you aware of any other projects who might be affected by this change? Unfortunately I couldn't find a way to convince the pypi download stats to give up such information or dep graphs... https://github.com/pypa/packaging-problems/issues/54

Reply:
I had confused this with h2, and your argument makes sense.